### ISSUE TITLE
user encoder

### ISSUE THREAD

**Issue by shainaraza** [wuch15/KDD-NPA#4]
Should I consider this as user encoder according to NPA
model = Model(candidates+all_news_input+[user_id], logits)

**Comment by wuch15** [comment]
This module actually includes both user encoder and click predictor. The output is the predicted click scores of "candidates".

**Comment by shainaraza** [comment]
thanks

**Comment by shainaraza** [comment]
and you are considering titles only , not the body of the news,

**Comment by shainaraza** [comment]
ideally, if I have to incorporate GRUs like in LSTUR, where should it be?
thanks in advance

**Comment by wuch15** [comment]
> and you are considering titles only , not the body of the news,

In this work, we only consider news titles, and news bodies are incorporated by our another work NAML. But you can simply concatenate news title and body in NPA.

> ideally, if I have to incorporate GRUs like in LSTUR, where should it be?

I think the method in LSTUR is not compatible with personalized attention. You may need to output each state in the GRU network first, and then apply personalized attention to the hidden state sequence.

**Comment by shainaraza** [comment]
thanks you very much
